File: HW04/131044061_CSE241_HW04/Triangle.cpp
#include<string>
#include <iostream>
#include "Triangle.h"
#include<cmath>
#include<cstdlib>
#include<fstream>
using namespace std;

Triangle::Triangle() {
	line = 0;
}

void Triangle::set_line(int l) {
	if(l>0){
		line = l;
	}
	else{
		cout<<"triangle olusmadi"<<endl;
	}
}

int Triangle::get_line() const {
	return line;
}

double Triangle::find_height() {
	height = sqrt(3)* (line/2);
	return height;
}

void Triangle::set_points(double p1_x, double p1_y, double p2_x,
		double p2_y, double p3_x, double p3_y) {
	double line1 = sqrt(((p1_x-p2_x)*(p1_x-p2_x)) + ((p1_y-p2_y)*(p1_y-p2_y)));
	double line2 = sqrt(((p1_x-p3_x)*(p1_x-p3_x)) + ((p1_y-p3_y)*(p1_y-p3_y)));
	double line3 = sqrt(((p3_x-p2_x)*(p3_x-p2_x)) + ((p3_y-p2_y)*(p3_y-p2_y)));
	if(line1<=line || line2<=line || line3<=line){
		points[0][0] = p1_x;
		points[0][1] = p1_y;
		points[1][0] = p2_x;
		points[1][1] = p2_y;
		points[2][0] = p3_x;
		points[2][1] = p3_y;
	}
	else{
		cout<<"triangle koordinantlar hatali"<<endl;
	}
}


double Triangle::get_point_x(int index) const {
	return points[index][0];
}

double Triangle::get_point_y(int index) const {
	return points[index][1];
}

const double Triangle::area() {
	return (double)get_line() * find_height() / 2;
}

const double Triangle::lenght() {
	return (double)get_line() * 3;
}

const bool Triangle::operator ==(Triangle& other) {
	return this->area() == other.area();
}

const bool Triangle::operator !=(Triangle& other) {
	return !(this->area() == other.area());
}

const Triangle Triangle::operator +(double num) {
	Triangle temp;
	temp.set_line(this->get_line() + num);
	return temp;
}

const Triangle Triangle::operator -(double num) {
	Triangle temp;
	temp.set_line(this->get_line() + num);
	return temp;
}

int Triangle::get_numOfShape() {
	numOfShape++;
	return numOfShape;
}

double Triangle::result_area(Triangle &tri) {
	return numOfShape * tri.area();
}

ofstream& operator <<(ofstream& ofs, const Triangle& tri) {
	ofs<<tri.get_idntfy()<<tri.get_point_x(0)<<" "<<tri.get_point_y(0)<<" "<<tri.get_point_x(1)
			<<" "<<tri.get_point_y(1)
			<<" "<<tri.get_point_x(2)<<" "<<tri.get_point_y(2)<<"' "<<tri.get_stroke()<<tri.fill<<endl;
	return ofs;
}

string Triangle::get_idntfy() const {
	return idntfy;
}

string Triangle::get_stroke() const {
	return stroke;
}

void Triangle::draw(ofstream & ofs) const {
	ofs<<*this;
}

int Triangle::reset_numOfShape() {
	numOfShape = 0;
	return numOfShape;
}

File: HW07/131044061_CSE241_HW07_caner_karakas/hw07/src/Triangle.java
import java.awt.Color;
import java.awt.Graphics;
import java.awt.Polygon;
import javax.swing.JPanel;
import java.lang.*;


public class Triangle extends JPanel implements Shape {

	private static final long serialVersionUID = 1L;
	private static int numOfShape = 0;
	private double line;
	private double height;
	private int [][] points = new int[6][6];
	
	/**
	 * default constructor
	 */
	public Triangle() {
	}
	/**
	 * take Triangle object constructor
	 * @param l
	 * @throws Exception
	 */
	public Triangle(Triangle l) throws Exception {
		this.set_line(l.get_line());
		this.set_coords(l.get_point_x(0), l.get_point_y(0), l.get_point_x(1), l.get_point_y(1), l.get_point_x(2), l.get_point_y(2));
	}
	/**
	 * take double line
	 * @param l
	 * @throws Exception
	 */
	public Triangle(double l) throws Exception {
		this.set_line(l);
	}
	
	@Override
	public int compareTo(Shape other) {
		if(this.area() == other.area())
			return 1;
		return 0;
	}

	@Override
	public double area() {return height * line / 2;}

	@Override
	public double perimeter() {return line * 3;}

	@Override
	public Shape increment() {
		for(int i = 0; i<6; i++){
			for(int j=0; j<6; j++)
				points[i][j]++;
		}
		return this;
	}

	@Override
	public Shape decrement() {
		for(int i = 0; i<6; i++){
			for(int j=0; j<6; j++)
				points[i][j]--;
		}
		return this;
	}

	@Override
	public void draw(Graphics graph) {
		this.paintComponent(graph);
	}
	
	@Override
	public void paintComponent(Graphics graph){
		super.paintComponent(graph);
		int [] xPoints = new int [3];
		int [] yPoints = new int [3];
		for(int i = 0; i<3; i++) {
			xPoints[i] = points[i][0];
		}
		for(int i = 0; i<3; i++) {
			yPoints[i] = points[i][1];
		}
		graph.setColor(Color.RED);
		graph.drawPolygon(xPoints, yPoints, 3);
	}
	
	/**
	 * set_line
	 * @param l
	 * @throws Exception
	 */
	public void set_line(double l) throws Exception{
		if(l>0){
			line = l;
			height = (line/2) * Math.sqrt(3);
		}
		else{/*
			System.out.printf("Error111!!");
			System.exit(0);*/
			new Exception();
		}
	}
	
	public double get_line(){return line;}
	
	public double get_height(){return height;}
	
	/**
	 * set_coords
	 * @param p1_x
	 * @param p1_y
	 * @param p2_x
	 * @param p2_y
	 * @param p3_x
	 * @param p3_y
	 * @throws Exception
	 */
	public void set_coords(double p1_x, double p1_y, double p2_x,
		double p2_y, double p3_x, double p3_y) throws Exception{
		double line1 = Math.sqrt(((p1_x-p2_x)*(p1_x-p2_x)) + ((p1_y-p2_y)*(p1_y-p2_y)));
		double line2 = Math.sqrt(((p1_x-p3_x)*(p1_x-p3_x)) + ((p1_y-p3_y)*(p1_y-p3_y)));
		double line3 = Math.sqrt(((p3_x-p2_x)*(p3_x-p2_x)) + ((p3_y-p2_y)*(p3_y-p2_y)));
		if(line1<=line || line2<=line || line3<=line){
			points[0][0] = (int) p1_x;
			points[0][1] = (int) p1_y;
			points[1][0] = (int) p2_x;
			points[1][1] = (int) p2_y;
			points[2][0] = (int) p3_x;
			points[2][1] = (int) p3_y;
		}
		else{/*
			System.out.printf("Error!!");
			System.exit(0);*/
			new Exception();
		}
	}
	
	public double get_point_x(int index) {return points[index][0];}
	
	public double get_point_y(int index) {return points[index][1];}
	
	static double result_area(Triangle tri){return numOfShape * tri.area();}
	
	static int get_numOfShape(){numOfShape++;return numOfShape;}
	
	static int reset_numOfShape(){numOfShape = 0;return numOfShape;}

}
